dask_kubernetes/operator/controller/controller.py

```diff
@@ -626,35 +630,121 @@ async def daskautoscaler_create(spec, name, namespace, logger, **kwargs):
         logger.info(f"Successfully adopted by {spec['cluster']}")
 
 
-@kopf.timer("daskautoscaler.kubernetes.dask.org", interval=5.0)
+@kopf.timer("daskautoscaler.kubernetes.dask.org", interval=30.0)
 async def daskautoscaler_adapt(spec, name, namespace, logger, **kwargs):
-    # Ask the scheduler for the desired number of worker
-    try:
-        desired_workers = await get_desired_workers(
-            scheduler_service_name=f"{spec['cluster']}-scheduler",
-            namespace=namespace,
-            logger=logger,
-        )
-    except SchedulerCommError:
-        logger.error("Unable to get desired number of workers from scheduler.")
-        return
+    async with kubernetes.client.api_client.ApiClient() as api_client:
+        coreapi = kubernetes.client.CoreV1Api(api_client)
 
-    # Ensure the desired number is within the min and max
-    desired_workers = max(spec["minimum"], desired_workers)
-    desired_workers = min(spec["maximum"], desired_workers)
+        pod_ready = False
+        try:
+            scheduler_pod = await coreapi.read_namespaced_pod(
+                f"{spec['cluster']}-scheduler", namespace
+            )
+            if scheduler_pod.status.phase == "Running":
+                pod_ready = True
+        except ApiException as e:
+            if e.status != 404:
+                raise e
+
+        if not pod_ready:
+            logger.info("Scheduler not ready, skipping autoscaling")
+            return
 
-    # Update the default DaskWorkerGroup
-    # TODO Only update if the value has changed
-    async with kubernetes.client.api_client.ApiClient() as api_client:
         customobjectsapi = kubernetes.client.CustomObjectsApi(api_client)
         customobjectsapi.api_client.set_default_header(
             "content-type", "application/merge-patch+json"
         )
-        await customobjectsapi.patch_namespaced_custom_object_scale(
+
+        autoscaler_resource = await customobjectsapi.get_namespaced_custom_object(
+            group="kubernetes.dask.org",
+            version="v1",
+            plural="daskautoscalers",
+            namespace=namespace,
+            name=name,
+        )
```

Comment:
It looks like the only attribute being accessed on this object is `metadata`. This is already available in `kwargs["metadata"]`. I recommend updating the function signature to include a `metadata` kwarg and access that directly.

Reply:
I notice this still hasn't been addressed. I'd like to remove this unnecessary HTTP call if possible. Was there a reason why you couldn't use `kwargs["metadata"]` here?

Reply:
Update on this @jacobtomlinson, it looks like we're getting `TypeError: daskautoscaler_adapt() missing 1 required positional argument: 'metadata'` with this setup.   I've double check that referencing correctly and it looks right to me so perhaps timers don't get the metadata of the resource?  May explain why I opted to do it this way in the first place, but can't remember :).